### ISSUE TITLE
Search Results do not show Place name sometimes (ericb)

### ISSUE THREAD

**Issue by FreeUKGenIssues** [FreeUKGen/FreeUKRegProductIssues#172]
Issue reported by **ericb** at 2015-02-16 15:17:25 UTC
Time: 2015-02-16T15:12:09+00:00
Session ID: 325b5b88c334795895a79e6349b83478
Problem Page URL: [/search_queries/54e2089fe93790ddd0000001?page_number=0](/search_queries/54e2089fe93790ddd0000001?page_number=0)
Previous Page URL: [http://freereg2.freereg.org.uk/search_records/54ce58c9e93790802f32a534?search_id=54e2089fe93790ddd0000001](http://freereg2.freereg.org.uk/search_records/54ce58c9e93790802f32a534?search_id=54e2089fe93790ddd0000001)
Reported Issue:
take the attached sample 
query was Fysher with soundex ON, start date + end date and ALL types.

Some show Place name correctly, others like the one viewed do not>
The Place name for the (St  Peter & StPaul) (Unspecified) is Bolton-By-Bowland if it is of help.    

![Screenshot](http://freereg2.freereg.org.uk/home/apache/hosts/freereg2/production/tmp/carrierwave/1424099845-17293-8302/Screen_shot_2015-02-16_at_15.12.24.jpg)

**Comment by Sherlock21** [comment]
Is this "feature" a function of whether the Register is Unspecified (= messes up the Place Name being shown) or not ( = Place name comes up OK) as it happened on other searches in Norfolk which I ran as a test.

**Comment by Sherlock21** [comment]
My Speculation above that its Register type = Unspecified which caused the issue is not true. it's just happened again with Register type = AT also.

This suggestion might just be a red herring!

**Comment by Sherlock21** [comment]
refer to the record attached, for Ann Hudson baptised 9 Oct 1712 at NFK/ Haveringland  - this has no register type shown if you navigate to the details page So how did that fail to get the Unspecified default?

**Comment by FreeUKGenIssues** [comment]
This is very interesting.  There's an algorithm that builds the text shown
in the Location column on the search results screen, that looks at place
name, church name, and register type.  It's a surprisingly tricky bit of
code, as it has to deal with making sure that the word-wrapping works
appropriately, as well as the sort order.

Apparently there's a problem with this code, so that unspecified register
types break the algorithm for place names.

Would you mind looking at the most recent version of the code at
http://test2.freereg.org.uk/ to see if the rebuild also has this problem?
The records on that site only contain transcripts from user IDs beginning
with letters a-e.

Ben

**Comment by Sherlock21** [comment]
Hi, Ben

A similar search on your test site, gave this result:-

I do not know who this transcriber was  in the above, from your test site.
But looking back at the first copy result page, for Edward Feazor 1602- the transcriber there was I. Marlborough so won't be in your test set.

But the error is still in there.

AND even worse: see the last record in this set attached. its by C Smallwood in FreeReg2.freereg.org site I am using and i think it looks like this same person in the middle of this list from freereg2.freereg.org done just now. HERE the place name comes out OK! 

Strange. Very Strange.

My id is ericb but I think ALL my transcriptions show the proper Register Type, so we will not have any unspecified ones.
I could fabricate a few if you wanted? So long as I can upload into your TEST site.

Eric

**Comment by benwbrum** [comment]
Eric, there's no need to fabricate the problem -- at this point I have
enough information to research and fix it.

Nice job to everyone involved -- this is a very good catch, which would
have been embarrassing to release.

**Comment by Sherlock21** [comment]
Ben - OK, i will leave you to it.

Shout if you need a "quiet" test done. 

meanwhile on another issue from John Pingram, I have copied you in to my email to him with my findings for his problem. Again, I hope there  are a few clues in that of use to you

snag numbers I think are #170 and #172 but dont take it as fact -i have cleared my desktop & shut all down for the evening now.
My turn to do the cooking!

Bye. 
Eric B

**Comment by benwbrum** [comment]
One good example of this is imarlb.WRYBBBBA2.CSV.537

**Comment by Sherlock21** [comment]
I just looked at the file listed above ( imarlb.WRYBBBBA2.csv ) I don't know why Ben chose this, but the version in freereg2.freereg.org.uk right now has an issue: It has the Place name spelled wrongly - the "By" in the middle should be in lower case to match the Place names table. [like the rest of the files for this place are correct in that respect]  SO is this why you found it, for the wrong reasons?
In the previous version of the Application these "InCorrect" files were flagged with a red error label, then later just the date updated came in red. Now none of this particular file seem to be listed 

I found it in my "Find a Specific File" & Downloaded it. 
The content looks file at first sight EXCEPT for the PlaceName spelling.
Nothing odd with #537 that I can see in text editor.

Is this change of "not now showing wrong place name files", another DIFFERNT BUG? or is it a change in intent.?
